internal/release/promote/pr_rendering.go

```diff
@@ -0,0 +1,200 @@
+package promote
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/nestoca/joy/api/v1alpha1"
+	"golang.org/x/mod/semver"
+)
+
+type ChangeType string
```

Comment:
There's generally no reason to use a semantic type like this. It gives us the feeling of an enum, but its just a lie. Constant strings would be just as good, and introduce less mental overhead.

If you want me to show you how to emulate enums in Go, we can do that offline. But generally its only good across package boundaries like when you are writing a lib for others to consume.

Reply:
As I mentioned above, change type is exposed to template, so I think it's a good thing to formalize it, along with allowed values. If you are strictly against that, please let's discuss, because I don't think I understand the rationale.